corehq/sql_db/connections.py

```diff
@@ -113,23 +112,68 @@ def get_connection_string(self, engine_id):
         return connection_string or self.db_connection_map['default']
 
     def _populate_connection_map(self):
-        self._add_django_db(ICDS_UCR_ENGINE_ID, 'ICDS_UCR_DATABASE_ALIAS')
-        self._add_django_db(ICDS_TEST_UCR_ENGINE_ID, 'ICDS_UCR_TEST_DATABASE_ALIAS')
+        reporting_db_config = self._get_reporting_db_config()
+        if not reporting_db_config:
+            self._populate_from_legacy_settings()
+        else:
+            for engine_id, db_config in reporting_db_config.items():
+                replicas = []
+                if isinstance(db_config, dict):
+                    db_alias = db_config['DJANGO_ALIAS']
+                    replicas = db_config.get('READ_REPLICAS', [])
+                else:
+                    db_alias = db_config
+                self._add_django_db(engine_id, db_alias)
+                self.replica_mapping[engine_id] = replicas
+                for replica in replicas:
+                    assert replica not in self.db_connection_map, replica
+                    self._add_django_db(replica, replica)
+
+        if DEFAULT_ENGINE_ID not in self.db_connection_map:
+            self._add_django_db(DEFAULT_ENGINE_ID, 'default')
+        if UCR_ENGINE_ID not in self.db_connection_map:
+            self._add_django_db(UCR_ENGINE_ID, 'default')
+
+        self._add_django_db_from_settings_key(ICDS_UCR_ENGINE_ID, 'ICDS_UCR_DATABASE_ALIAS')
+        self._add_django_db_from_settings_key(ICDS_TEST_UCR_ENGINE_ID, 'ICDS_UCR_TEST_DATABASE_ALIAS')
```

Comment:
Should also deprecate the "test" ucr engine as now it just points to the real thing

Reply:
:+1: don't want to complicate this PR - I think it will take a bit of untangling to remove that since there are a bunch of reports that still point to it.

Reply:
Yeah. I'm going to wait until the dev on those reports slows down before doing that to avoid conflicts